doc/source/tutorial/stats/sampling_pinv.rst

```diff
@@ -0,0 +1,271 @@
+    >>> plt.xlabel("x")
+    >>> plt.ylabel("PDF(x)")
+    >>> plt.title("Samples drawn using PINV method.")
+    >>> plt.legend()
+    >>> plt.show()
+
+The maximum tolerated error (i.e. u-resolution) can be changed by passing the
+``u_resolution`` keyword during initialization:
+
+    >>> rng = NumericalInversePolynomial(dist, u_resolution=1e-12,
+    ...                                  random_state=urng)
+
+This leads to a more accurate approximation of the PPF and the
+generated RVs follow the exact distribution more closely. Although, note
+that it comes at the cost of an expensive setup.
+
+The setup time mainly depends on the number of times the PDF is evaluated.
+It is more costly for PDFs that are difficult to evaluate. Note that we can
+ignore the normalization constant to speed up the evaluations of the PDF.
+PDF evaluations increase during setup for small values of ``u_resolution``.
+
+    >>> from scipy.stats import NumericalInversePolynomial
+    >>> class StandardNormal:
+    ...     def __init__(self):
+    ...         self.callbacks = 0
+    ...     def pdf(self, x):
+    ...         self.callbacks += 1
+    ...         return np.exp(-0.5 * x*x)
+    ... 
+    >>> dist = StandardNormal()
+    >>> urng = np.random.default_rng()
+    >>> # u_resolution = 10^-8
+    >>> # => less PDF evaluations required
+    >>> # => faster setup
+    >>> rng = NumericalInversePolynomial(dist, u_resolution=1e-8,
+    ...                                  random_state=urng)
+    >>> dist.callbacks
+    4095
+    >>> dist.callbacks = 0  # reset the number of callbacks
+    >>> # u_resolution = 10^-10 (default)
+    >>> # => more PDF evaluations required
+    >>> # => slow setup
+    >>> rng = NumericalInversePolynomial(dist, u_resolution=1e-10,
+    ...                                  random_state=urng)
+    >>> dist.callbacks
+    11454
+    >>> dist.callbacks = 0  # reset the number of callbacks
+    >>> # u_resolution = 10^-12
+    >>> # => lots of PDF evaluations required
+    >>> # => very slow setup
+    >>> rng = NumericalInversePolynomial(dist, u_resolution=1e-12,
+    ...                                  random_state=urng)
+    13902
+
+As we can see, the number of PDF evaluations required is very high and a
+fast PDF is critical to the algorithm. Though, this helps reduce the number
+of subintervals required to achieve the error goal which saves memory and
+makes sampling fast. `NumericalInverseHermite` is a similar inversion method
+that inverts the CDF based on Hermite interpolation and provides control
+over the maximum tolerated error via u-resolution. But it makes use of a lot
+more intervals compared to `NumericalInversePolynomial`:
+
+    >>> from scipy.stats import NumericalInverseHermite
+    >>> # NumericalInverseHermite accepts a `tol` parameter to set the
+    >>> # u-resolution of the generator.
+    >>> rng_hermite = NumericalInverseHermite(norm(), tol=1e-12)
```

Comment:
i just noted the slight inconsistency in the interface:

tol vs u_resolution
intervals vs n_intervals

just mentioning it here to start a discussion

Reply:
> tol vs u_resolution
> intervals vs n_intervals

I also wanted to discuss this: `u_resolution` is much more descriptive than `tol` and it aligns with the naming convention of the paper and UNU.RAN. So, should we deprecate `tol` and use `u_resolution` instead? What do you think, @mdhaber?

I will change `n_intervals` to just `intervals` as both are descriptive enough.

Reply:
Changed `n_intervals` to just `intervals`.

Reply:
I am also in favour of u_resolution

Reply:
Let me know if you agree, @mdhaber. If you do, I think this is good to go and we can start working on shifting `NumericalInverseHermite` to UNU.RAN.

Reply:
I'm not really convinced, but I'm not blocking anything. I'm that case, when you work on `NumericalInverseHermite`, consider deprecating `tol` and accepting `u_resolution`, then.

Reply:
ok, looks like we can proceed with `u_resolution` for PINV. I would also prefer to change that in `NumericalInverseHermite` (`tol` is not a bad option either). Compared to  `random_state` discussion, this is a very small change that is easy to implement.